Auto update failing for sops

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [sops](https://github.com/termux/termux-packages/blob/6ba1bdc5ef7171757ea2403967359eaa4393ddff/packages/sops) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="console">
INFO: Updating sops [Current version: 3.8.1]
INFO: package being updated to 3.9.0.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  237k    0  237k    0     0   527k      0 --:--:-- --:--:-- --:--:--  527k

You are about to commit these changes:

--------------------
diff --git a/packages/sops/build.sh b/packages/sops/build.sh
index 0cad1be3d..28fd2796a 100644
--- a/packages/sops/build.sh
+++ b/packages/sops/build.sh
@@ -2,9 +2,9 @@ TERMUX_PKG_HOMEPAGE=https://github.com/mozilla/sops
 TERMUX_PKG_DESCRIPTION="Simple and flexible tool for managing secrets"
 TERMUX_PKG_LICENSE="MPL-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="3.8.1"
+TERMUX_PKG_VERSION="3.9.0"
 TERMUX_PKG_SRCURL="https://github.com/mozilla/sops/archive/v${TERMUX_PKG_VERSION}.tar.gz"
-TERMUX_PKG_SHA256=5ca70fb4f96797d09012c705a5bb935835896de7bcd063b98d498912b0e645a0
+TERMUX_PKG_SHA256=eda01428a4178903b2d9552940fe441d93fab7b4582cd3f5fee7b6b73404d8cb
 TERMUX_PKG_AUTO_UPDATE=true
 
 termux_step_make_install() {
--------------------

bump(main/sops): 3.9.0


Not committing to Git!

INFO: Trying to build package.
INFO: Package sops exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building sops for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13134  100 13134    0     0  31310      0 --:--:-- --:--:-- --:--:-- 31346
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   2007      0 --:--:-- --:--:-- --:--:--  2007
100   833  100   833    0     0   2006      0 --:--:-- --:--:-- --:--:--  2002
gpg: Signature made Fri 28 Jun 2024 09:56:42 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13320  100 13320    0     0  31787      0 --:--:-- --:--:-- --:--:-- 31866
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   2032      0 --:--:-- --:--:-- --:--:--  2036
gpg: Signature made Mon 24 Jun 2024 02:28:06 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13120  100 13120    0     0  31475      0 --:--:-- --:--:-- --:--:-- 31538
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   4490      0 --:--:-- --:--:-- --:--:--  4502
gpg: Signature made Thu 27 Jun 2024 11:42:17 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://github.com/mozilla/sops/archive/v3.9.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  237k    0  237k    0     0   792k      0 --:--:-- --:--:-- --:--:--  792k
go: downloading github.com/fatih/color v1.17.0
go: downloading github.com/goware/prefixer v0.0.0-20160118172347-395022866408
go: downloading github.com/mitchellh/go-wordwrap v1.0.1
go: downloading github.com/sirupsen/logrus v1.9.3
go: downloading golang.org/x/net v0.26.0
go: downloading filippo.io/age v1.2.0
go: downloading github.com/lib/pq v1.10.9
go: downloading github.com/pkg/errors v0.9.1
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/golang/protobuf v1.5.4
go: downloading google.golang.org/grpc v1.64.0
go: downloading google.golang.org/protobuf v1.34.2
go: downloading github.com/ProtonMail/go-crypto v1.1.0-alpha.3-proton
go: downloading github.com/getsops/gopgagent v0.0.0-20240527072608-0c14999532fe
go: downloading golang.org/x/term v0.21.0
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading github.com/mattn/go-isatty v0.0.20
go: downloading golang.org/x/sys v0.21.0
go: downloading golang.org/x/crypto v0.24.0
go: downloading github.com/Azure/azure-sdk-for-go/sdk/azcore v1.12.0
go: downloading github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.7.0
go: downloading github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azkeys v1.1.0
go: downloading cloud.google.com/go/kms v1.18.0
go: downloading google.golang.org/api v0.186.0
go: downloading cloud.google.com/go v0.115.0
go: downloading github.com/hashicorp/vault/api v1.14.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/aws/aws-sdk-go-v2 v1.30.0
go: downloading github.com/aws/aws-sdk-go-v2/config v1.27.21
go: downloading github.com/aws/aws-sdk-go-v2/credentials v1.17.21
go: downloading github.com/aws/aws-sdk-go-v2/service/kms v1.34.1
go: downloading github.com/aws/aws-sdk-go-v2/service/sts v1.29.1
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20240624140628-dc46fd24d27d
go: downloading google.golang.org/genproto v0.0.0-20240624140628-dc46fd24d27d
go: downloading github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/internal v1.0.1
go: downloading github.com/Azure/azure-sdk-for-go/sdk/internal v1.9.0
go: downloading github.com/AzureAD/microsoft-authentication-library-for-go v1.2.2
go: downloading github.com/cenkalti/backoff/v3 v3.2.2
go: downloading github.com/go-jose/go-jose/v4 v4.0.2
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/hashicorp/go-cleanhttp v0.5.2
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/hashicorp/go-retryablehttp v0.7.7
go: downloading github.com/hashicorp/go-rootcerts v1.0.2
go: downloading github.com/hashicorp/go-secure-stdlib/parseutil v0.1.8
go: downloading github.com/hashicorp/go-secure-stdlib/strutil v0.1.2
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/mitchellh/mapstructure v1.5.0
go: downloading golang.org/x/time v0.5.0
go: downloading cloud.google.com/go/iam v1.1.8
go: downloading cloud.google.com/go/longrunning v0.5.7
go: downloading github.com/googleapis/gax-go/v2 v2.12.5
go: downloading google.golang.org/genproto/googleapis/api v0.0.0-20240624140628-dc46fd24d27d
go: downloading github.com/aws/smithy-go v1.20.2
go: downloading github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.16.8
go: downloading github.com/aws/aws-sdk-go-v2/service/sso v1.21.1
go: downloading github.com/aws/aws-sdk-go-v2/internal/ini v1.8.0
go: downloading github.com/aws/aws-sdk-go-v2/service/ssooidc v1.25.1
go: downloading github.com/aws/aws-sdk-go-v2/internal/configsources v1.3.12
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/accept-encoding v1.11.2
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.11.14
go: downloading github.com/cloudflare/circl v1.3.9
go: downloading github.com/google/uuid v1.6.0
go: downloading github.com/kylelemons/godebug v1.1.0
go: downloading github.com/pkg/browser v0.0.0-20240102092130-5ac0b6a4141c
go: downloading github.com/hashicorp/go-sockaddr v1.0.6
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading cloud.google.com/go/auth v0.6.0
go: downloading golang.org/x/oauth2 v0.21.0
go: downloading cloud.google.com/go/auth/oauth2adapt v0.2.2
go: downloading cloud.google.com/go/compute/metadata v0.3.0
go: downloading go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.52.0
go: downloading go.opencensus.io v0.24.0
go: downloading go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.52.0
go: downloading github.com/aws/aws-sdk-go-v2/internal/endpoints/v2 v2.6.12
go: downloading github.com/golang-jwt/jwt/v5 v5.2.1
go: downloading golang.org/x/text v0.16.0
go: downloading github.com/google/s2a-go v0.1.7
go: downloading go.opentelemetry.io/otel v1.27.0
go: downloading go.opentelemetry.io/otel/metric v1.27.0
go: downloading go.opentelemetry.io/otel/trace v1.27.0
go: downloading github.com/googleapis/enterprise-certificate-proxy v0.3.2
go: downloading github.com/felixge/httpsnoop v1.0.4
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading github.com/go-logr/logr v1.4.2
go: downloading golang.org/x/sync v0.7.0
go: downloading github.com/go-logr/stdr v1.2.2
go: module github.com/golang/protobuf is deprecated: Use the "google.golang.org/protobuf" module instead.
GOPROXY=https://proxy.golang.org go install github.com/getsops/sops/v3/cmd/sops
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading github.com/urfave/cli v1.22.15
go: downloading github.com/blang/semver v3.5.1+incompatible
go: downloading gopkg.in/ini.v1 v1.67.0
go: downloading cloud.google.com/go/storage v1.42.0
go: downloading github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.17.1
go: downloading github.com/aws/aws-sdk-go-v2/service/s3 v1.56.1
go: downloading github.com/google/go-cmp v0.6.0
go: downloading github.com/cpuguy83/go-md2man/v2 v2.0.4
go: downloading github.com/russross/blackfriday/v2 v2.1.0
go: downloading github.com/jmespath/go-jmespath v0.4.0
go: downloading github.com/aws/aws-sdk-go-v2/internal/v4a v1.3.12
go: downloading github.com/aws/aws-sdk-go-v2/aws/protocol/eventstream v1.6.2
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/checksum v1.3.14
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/s3shared v1.17.12
# github.com/hashicorp/go-sockaddr
../../../pkg/mod/github.com/hashicorp/go-sockaddr@v1.0.6/route_info_android.go:8:6: routeInfo redeclared in this block
	../../../pkg/mod/github.com/hashicorp/go-sockaddr@v1.0.6/route_info.go:19:6: other declaration of routeInfo
../../../pkg/mod/github.com/hashicorp/go-sockaddr@v1.0.6/route_info_default.go:11:6: NewRouteInfo redeclared in this block
	../../../pkg/mod/github.com/hashicorp/go-sockaddr@v1.0.6/route_info_android.go:14:6: other declaration of NewRouteInfo
../../../pkg/mod/github.com/hashicorp/go-sockaddr@v1.0.6/route_info_default.go:17:21: method routeInfo.GetDefaultInterfaceName already declared at ../../../pkg/mod/github.com/hashicorp/go-sockaddr@v1.0.6/route_info_android.go:22:21
make: *** [Makefile:36: install] Error 1
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2024-06-28 12:14:35 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/9712480079">9712480079</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/sops/build.sh
+++ b/packages/sops/build.sh
@@ -1,20 +1,28 @@
-TERMUX_PKG_HOMEPAGE=https://github.com/mozilla/sops
+TERMUX_PKG_HOMEPAGE=https://github.com/getsops/sops
 TERMUX_PKG_DESCRIPTION="Simple and flexible tool for managing secrets"
 TERMUX_PKG_LICENSE="MPL-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="3.8.1"
-TERMUX_PKG_SRCURL="https://github.com/mozilla/sops/archive/v${TERMUX_PKG_VERSION}.tar.gz"
-TERMUX_PKG_SHA256=5ca70fb4f96797d09012c705a5bb935835896de7bcd063b98d498912b0e645a0
+TERMUX_PKG_VERSION="3.9.0"
+TERMUX_PKG_SRCURL=https://github.com/getsops/sops/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=eda01428a4178903b2d9552940fe441d93fab7b4582cd3f5fee7b6b73404d8cb
 TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_UPDATE_TAG_TYPE="latest-release-tag"
 
 termux_step_make_install() {
 	termux_setup_golang
 
 	export GOPATH="${TERMUX_PKG_BUILDDIR}"
-	mkdir -p "${GOPATH}/src/go.mozilla.org"
-	cp -a "${TERMUX_PKG_SRCDIR}" "${GOPATH}/src/go.mozilla.org/sops"
-	cd "${GOPATH}/src/go.mozilla.org/sops" || return 9
+	mkdir -p "${GOPATH}/src/github.com/getsops"
+	cp -a "${TERMUX_PKG_SRCDIR}" "${GOPATH}/src/github.com/getsops/sops"
+	cd "${GOPATH}/src/github.com/getsops/sops" || return 9
 	go get -d -v
+
+	local d
+	for d in ${GOPATH}/pkg/mod/github.com/hashicorp/go-sockaddr*; do
+		chmod +w -R "${d}"
+		patch --silent -p1 -d "${d}" < "$TERMUX_PKG_BUILDER_DIR/go-sockaddr.diff" || :
+	done
+
 	make install
 
 	install -Dm700 "${GOPATH}/bin/"*/sops "${TERMUX_PREFIX}/bin/sops"

--- a/packages/sops/go-sockaddr.diff
+++ b/packages/sops/go-sockaddr.diff
@@ -1,0 +1,40 @@
+diff --git a/route_info_android.go b/route_info_android.go
+index 9885915a..94c25789 100644
+--- a/route_info_android.go
++++ b/route_info_android.go
+@@ -5,10 +5,6 @@ import (
+ 	"os/exec"
+ )
+
+-type routeInfo struct {
+-	cmds map[string][]string
+-}
+-
+ // NewRouteInfo returns a Android-specific implementation of the RouteInfo
+ // interface.
+ func NewRouteInfo() (routeInfo, error) {
+diff --git a/route_info_default.go b/route_info_default.go
+index 6df864ba..db0052fa 100644
+--- a/route_info_default.go
++++ b/route_info_default.go
+@@ -1,5 +1,5 @@
+-//go:build android || nacl || plan9 || js
+-// +build android nacl plan9 js
++//go:build nacl || plan9 || js
++// +build nacl plan9 js
+
+ package sockaddr
+
+@@ -8,12 +8,3 @@ func getDefaultIfName() (string, error) {
+ 	return "", ErrNoInterface
+ }
+
+-func NewRouteInfo() (routeInfo, error) {
+-	return routeInfo{}, ErrNoRoute
+-}
+-
+-// GetDefaultInterfaceName returns the interface name attached to the default
+-// route on the default interface.
+-func (ri routeInfo) GetDefaultInterfaceName() (string, error) {
+-	return "", ErrNoInterface
+-}

